### ISSUE TITLE
NPC:NavSetGoalTarget( target, offfset ) not working

### ISSUE THREAD

**Issue by StrawWagen** [Facepunch/garrysmod-issues#5209]
### Details
NPC:NavSetGoalTarget( target, offfset ) always fails path generation ( throws false )

### Steps to reproduce
I set up a function in a local multiplayer x64 branch environment.

local function npcPathToEnt( me, target )
    me:SetSchedule( SCHED_ALERT_WALK )
    success = me:NavSetGoalTarget( target, Vector() )
    print( success, me, target )
    
end

This always prints false, me, and target
The map I tested this on has AI nodes, verified with ai_show_connect, and me/target are within 500 units of each other.

**Comment by robotboy655** [comment]
Yeah after looking into it further this function is completely useless and is not meant for what you want to achieve. It's basically used in the engine to update already active paths. I don't know why it was added in this form.

I made some changes on `dev` beta:
* Added NPC.NavSetGoalPos( pos )
* Changed NPC.NavSetGoalTarget - second argument defaults to 0,0,0; sets correct goal type for the path

That should fix it, at least when used with example from https://wiki.facepunch.com/gmod/NPC:NavSetRandomGoal
